movingpandas/trajectory.py

```diff
@@ -812,29 +902,53 @@ def _compute_distance(self, row, units):
         if not isinstance(pt0, Point):
             return 0.0
         if not isinstance(pt1, Point):
-            raise ValueError(
-                "Invalid trajectory! Got {} instead of point!".format(pt1))
+            raise ValueError("Invalid trajectory! Got {} instead of point!".format(pt1))
         if pt0 == pt1:
             return 0.0
         # If no units declared, return distance in m (geographic) or CRS units
-        if units == "default":
+        if units.distance is None:
             if self.is_latlon:
                 dist_computed = measure_distance_geodesic(pt0, pt1)
             else:  # The following distance will be in CRS units that might not be meters!
                 dist_computed = measure_distance_euclidean(pt0, pt1)
         # If units declared, return distance in those units if CRS units known
         # Assumes CRS in meters if CRS units unknown
-        if units in [d["abbr"] for d in DISTANCE_UNIT_LIST]:
+        if units.distance in [d["abbr"] for d in DISTANCE_UNIT_LIST]:
             if self.is_latlon:
-                dist_computed = measure_distance_geodesic(
-                    pt0, pt1)/[d["conv"] for d in DISTANCE_UNIT_LIST if d.get("abbr") == units][0]
+                dist_computed = (
+                    measure_distance_geodesic(pt0, pt1)
+                    / [
```

Comment:
This section is hard to understand. Maybe put the code to get the conversion factor into a helper function.

Reply:
A helper function will also help avoid code duplication. Currently, this statement appears multiple time throughout the code.

Reply:
Yes you're right, this is a much better way! Much less duplication, much more Pythonic too I think.